src/api/app/models/project.rb

```diff
@@ -1455,6 +1455,15 @@ def self.parent_projects(project_name)
     projects
   end
 
+  def lock(comment = nil)
+    self.transaction do
+      f = self.flags.find_by_flag_and_status('lock', 'disable')
```

Comment:
On Dienstag, 20. Oktober 2015, 03:49:13 CEST wrote Henne Vogelsang:
> > @@ -1455,6 +1455,15 @@ def self.parent_projects(project_name)
> >      projects
> >    end
> >  
> > +  def lock(comment = nil)
> > +    self.transaction do
> > +      f = self.flags.find_by_flag_and_status('lock', 'disable')
> 
> Why don't you just change the content of `status`?

also an option, but then we have a condition to create the new object or not.
it won't be simpler IMHO.

-- 

Adrian Schroeter
email: adrian@suse.de

SUSE Linux GmbH, GF: Felix Imend�rffer, Jane Smithard, Graham Norton, HRB 21284 (AG N�rnberg) 
 
Maxfeldstra�e 5                         
90409 N�rnberg 
Germany